root/static/scripts/relationship-editor/generic.js

```diff
@@ -203,7 +203,10 @@ const RE = MB.relationshipEditor = MB.relationshipEditor || {};
             pushInput(prefix, "link_type_id", editData.linkTypeID || "");
 
             if (relationship.getLinkType().orderable_direction !== 0) {
-                if (relationship.added() || changeData.linkOrder) {
+                if (relationship.added() ||
+                    changeData.linkOrder ||
+                    // Otherwise 0 is falsy and order can't be removed
+                    changeData.linkOrder === 0) {
```

Comment:
Nice catch. I've run into this a few times and generally use root/static/scripts/common/utility/nonEmpty.js (it simply checks that the value isn't null, undefined, or an empty string), so that might be ideal here.